src/control/destination/mod.rs

```diff
@@ -93,8 +93,8 @@ pub struct Resolution<B> {
 /// Metadata describing an endpoint.
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Metadata {
-    /// A set of Prometheus metric labels describing the destination.
-    dst_labels: Option<DstLabels>,
+    /// Arbitrary endpoint labels. Useful for telemetry, especially.
```

Comment:
This is definitely a nit, sorry, but I find "Useful for telemetry, especially" a bit awkward --- if the labels are basically treated as opaque by the proxy and can contain arbitrary labels, aren't they pretty much useful _only_ for telemetry?

Reply:
The `Metadata` type doesn't dictate how labels are used. They will be put into telemetry. However, there's nothing that prevents middlewares from peaking into the labels to make other decisions.